scheduler/reconcile.go

```diff
@@ -355,18 +355,12 @@ func (a *allocReconciler) computeGroup(group string, all allocSet) bool {
 
 	// Find delays for any lost allocs that have stop_after_client_disconnect
 	lostLater := lost.delayByStopAfterClientDisconnect()
-	rescheduleLater = append(rescheduleLater, lostLater...)
+	a.handleDelayedLost(lostLater, all, tg.Name)
 
 	// Create batched follow up evaluations for allocations that are
 	// reschedulable later and mark the allocations for in place updating
 	a.handleDelayedReschedules(rescheduleLater, all, tg.Name)
 
-	// Allocs that are lost and delayed have an attributeUpdate that correctly links to
-	// the eval, but incorrectly has the current (running) status
-	for _, d := range lostLater {
-		a.result.attributeUpdates[d.allocID].SetStop(structs.AllocClientStatusLost, structs.AllocClientStatusLost)
-	}
```

Comment:
yes, status is handled sort of twice in the planner, the alloc needs to be marked with the correct status but also sent to the planner in the `NodeUpdate` collection not the `NodeAllocation` collection. `attributeUpdates` all get added to the `NodeAllocation` part of the planner, which is wrong for our purposes. There's some followup to this behavior in https://github.com/hashicorp/nomad/pull/8105, which clarifies how the status gets applied.

I've been using this graph to follow the code: https://github.com/langmartin/nomad-dev/blob/master/doc/delayed-reschedules.svg. That's a hand-drawn graph, so it may have errors. red is control flow and green is data flow.